src/legacy/core_plugins/tile_map/public/components/wms_internal_options.tsx

```diff
@@ -64,7 +53,7 @@ function WmsInternalOptions({ wms, setValue }: WmsInternalOptions) {
     </EuiScreenReaderOnly>
   );
 
-  const setOptions: VisOptionsSetValue = (paramName, value) =>
+  const setOptions = (paramName: string, value: unknown) =>
```

Comment:
**nit:** I'm not sure if such strict typing makes much sense. We can refine this interface, but the code becomes less readable..
```suggestion
  const setOptions: <T extends keyof TileMapVisParams['wms']['options']>(
    paramName: T,
    value: TileMapVisParams['wms']['options'][T]
  ) => void = (paramName, value) =>
```